platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/gossip/chatter/ChatterGossip.java

```diff
@@ -0,0 +1,448 @@
+import com.swirlds.platform.Crypto;
+import com.swirlds.platform.FreezeManager;
+import com.swirlds.platform.PlatformConstructor;
+import com.swirlds.platform.StartUpEventFrozenManager;
+import com.swirlds.platform.components.CriticalQuorum;
+import com.swirlds.platform.components.CriticalQuorumImpl;
+import com.swirlds.platform.components.EventCreationRules;
+import com.swirlds.platform.components.EventMapper;
+import com.swirlds.platform.components.state.StateManagementComponent;
+import com.swirlds.platform.crypto.CryptoStatic;
+import com.swirlds.platform.event.EventCreatorThread;
+import com.swirlds.platform.event.EventIntakeTask;
+import com.swirlds.platform.event.GossipEvent;
+import com.swirlds.platform.event.creation.AncientParentsRule;
+import com.swirlds.platform.event.creation.BelowIntCreationRule;
+import com.swirlds.platform.event.creation.ChatterEventCreator;
+import com.swirlds.platform.event.creation.ChatteringRule;
+import com.swirlds.platform.event.creation.LoggingEventCreationRules;
+import com.swirlds.platform.event.creation.OtherParentTracker;
+import com.swirlds.platform.event.creation.StaticCreationRules;
+import com.swirlds.platform.event.intake.ChatterEventMapper;
+import com.swirlds.platform.event.linking.EventLinker;
+import com.swirlds.platform.gossip.AbstractGossip;
+import com.swirlds.platform.gossip.FallenBehindManagerImpl;
+import com.swirlds.platform.gossip.chatter.communication.ChatterProtocol;
+import com.swirlds.platform.gossip.chatter.config.ChatterConfig;
+import com.swirlds.platform.gossip.chatter.protocol.ChatterCore;
+import com.swirlds.platform.gossip.chatter.protocol.peer.PeerInstance;
+import com.swirlds.platform.gossip.shadowgraph.ShadowGraph;
+import com.swirlds.platform.gossip.shadowgraph.ShadowGraphSynchronizer;
+import com.swirlds.platform.metrics.EventIntakeMetrics;
+import com.swirlds.platform.network.communication.NegotiationProtocols;
+import com.swirlds.platform.network.communication.NegotiatorThread;
+import com.swirlds.platform.network.communication.handshake.VersionCompareHandshake;
+import com.swirlds.platform.observers.EventObserverDispatcher;
+import com.swirlds.platform.reconnect.DefaultSignedStateValidator;
+import com.swirlds.platform.reconnect.ReconnectController;
+import com.swirlds.platform.reconnect.ReconnectProtocol;
+import com.swirlds.platform.reconnect.emergency.EmergencyReconnectProtocol;
+import com.swirlds.platform.state.EmergencyRecoveryManager;
+import com.swirlds.platform.state.SwirldStateManager;
+import com.swirlds.platform.state.signed.SignedState;
+import com.swirlds.platform.threading.PauseAndClear;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+import org.apache.commons.lang3.tuple.Pair;
+
+/**
+ * Gossip implemented with the chatter protocol.
+ */
+public class ChatterGossip extends AbstractGossip {
```

Comment:
Would we call this `OutOfOrderGossip`?

Reply:
This is a tricky one... all of our implementations currently call it chatter. Let me have a conversation with Kelly on this, will leave this comment open until it's resolved one way or another.

Reply:
Talked with Kelly. We already call it chatter all over the place in the code, and we can continue to do so for the time being.